hack/build-carvel-packages.sh

```diff
@@ -0,0 +1,435 @@
+#!/usr/bin/env bash
+
+# Copyright 2020-2023 the Pinniped contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# This script can be used in conjunction with prepare-for-integration-tests.sh.
```

Comment:
```bash
# to this
PINNIPED_USE_LOCAL_KIND_REGISTRY=1 \
   ./hack/prepare-for-integration-tests.sh \
       --clean \
        --pre-install ./hack/build-carvel.sh \  # this will build the packages and then install the PackageRepository (1st half of script)
      --alternative_deploy ./hack/install-carvel.sh # this would install each package via PackageInstall files (called 3 times)
```

And then eliminate:

```bash
# unnecessary specific flags
--alternative_deploy_concierge
--alternative_deploy_supervisor
--alternative_deploy_local-user-authenticator
```

Reply:
For clarification, this means splitting this script in half.
The "build" stage, that can be passed to a new `pre-deploy` hook.
Then the "alternative-deploy" stage that deploys the PackageRepository,Package to the cluster.
We are accepting that we will rebuild & deploy the PackageRepository multiple times, since we call the script 3 times, it will build up to having all packages available by the final call.

Reply:
I thought we were talking about building all 3 packages and the package repository in the "pre-deploy" stage, and deploying the package repository in the "pre-deploy" stage? Then the "alternative-deploy" script only needs to be able to deploy one package install at a time.

Reply:
I may have forgotten that we landed on that, vs "its ok to apply as a noop multiple times" for the package install.  I'll move the package install and see if that speeds things up.

```bash
# will look something like this:
# the --pre-install being build-and-deploy and the 
# --alternate-deploy being install is a little quirky for names,
# but will be more accurate.
PINNIPED_USE_LOCAL_KIND_REGISTRY=1 ./hack/prepare-for-integration-tests.sh \
  --clean \
  --pre-install ./hack/lib/carvel_packages/build_and_deploy.sh \
  --alternate-deploy ./hack/lib/carvel_packages/install.sh
```